server/src-lib/Hasura/Server/App.hs

```diff
@@ -526,6 +514,30 @@ mkWaiApp isoLevel logger sqlGenCtx enableAL pool ci httpManager mode corsCfg ena
     getTimeMs :: IO Int64
     getTimeMs = (round . (* 1000)) `fmap` getPOSIXTime
 
+    initialiseSchemaCache :: m (E.PlanCache, SchemaCacheRef, Maybe UTCTime)
```

Comment:
Can we rename this to reflect that this also migrates catalog (if required)? Or split the catalog migration into a different top-level function?

The reason being at some point (in near future) we might want to run a check if catalog migration is required (using a different txiso level) and then skip it (that will solve #660)

Reply:
Renamed to `migrateAndInitialiseSchemaCache`.